qcodes/instrument/parameter.py

```diff
@@ -1387,6 +1388,30 @@ def timestamp(self) -> Optional[datetime]:
                 return None
             return self._parameter.source.cache.timestamp
 
+        @property
+        def valid(self) -> bool:
```

Comment:
shouldn;t the delegate cache totally forward this to the source parameter? and if there's no source parameter, then just be "valid" all the time? yeah, the meaning of "valid" is not clear when there's no source parameter... perhaps "valid" is a safer option for now

Reply:
Yes not sure why I did it like this originally